sw/device/lib/dif/dif_alert_handler.c

```diff
@@ -4,9 +4,942 @@
 
 #include "sw/device/lib/dif/dif_alert_handler.h"
 
+#include "sw/device/lib/base/bitfield.h"
+
 #include "alert_handler_regs.h"  // Generated.
 
 _Static_assert(ALERT_HANDLER_PARAM_N_CLASSES == 4,
                "Expected four alert classes!");
+_Static_assert(ALERT_HANDLER_PARAM_N_ESC_SEV == 4,
+               "Expected four escalation signals classes!");
+
+dif_alert_handler_result_t dif_alert_handler_init(
+    dif_alert_handler_params_t params, dif_alert_handler_t *handler) {
+  if (handler == NULL) {
+    return kDifAlertHandlerBadArg;
+  }
+
+  // We currently don't support more than 16 alerts correctly.
```

Comment:
We have 14 alerts in the system at this time, but this number is bound to grow since we are adding new IP and alerts to existing blocks. I would expect that this is going to be in a similar order of magnitude as the #IRQs.

Reply:
The only thing that changes when more alerts are added are the number of fields in the
```
ALERT_HANDLER.ALERT_EN
ALERT_HANDLER.ALERT_CLASS
ALERT_HANDLER.ALERT_CAUSE
```
CSRs. 

So we have to make sure that we can access the fields in those CSRs in a parametric way, even when they start to spill over into multiple CSRs (this happens at 32 alerts for `ALERT_EN` and `ALERT_CAUSE`, and 16 alerts for `ALERT_CLASS`).